openmmapi/src/NoseHooverIntegrator.cpp

```diff
@@ -0,0 +1,289 @@
+        );
+    }
+    noseHooverChains.at(chainID).setDefaultTemperature(temperature);
+
+}
+
+double NoseHooverIntegrator::getRelativeTemperature(int chainID) const {
+    if (chainID >= noseHooverChains.size()) {
+        throw OpenMMException("Cannot get relative temperature for chainID " + std::to_string(chainID)
+                + ". Only " + std::to_string(noseHooverChains.size()) + " have been registered with this integrator."
+        );
+    }
+    return noseHooverChains.at(chainID).getDefaultRelativeTemperature();
+}
+
+void NoseHooverIntegrator::setRelativeTemperature(double temperature, int chainID){
+    if (chainID >= noseHooverChains.size()) {
+        throw OpenMMException("Cannot set relative temperature for chainID " + std::to_string(chainID)
+                + ". Only " + std::to_string(noseHooverChains.size()) + " have been registered with this integrator."
+        );
+    }
+    noseHooverChains.at(chainID).setDefaultRelativeTemperature(temperature);
+
+}
+
+double NoseHooverIntegrator::getCollisionFrequency(int chainID) const {
+    if (chainID >= noseHooverChains.size()) {
+        throw OpenMMException("Cannot get collision frequency for chainID " + std::to_string(chainID)
+                + ". Only " + std::to_string(noseHooverChains.size()) + " have been registered with this integrator."
+        );
+    }
+    return noseHooverChains.at(chainID).getDefaultCollisionFrequency();
+}
+
+void NoseHooverIntegrator::setCollisionFrequency(double frequency, int chainID){
+    if (chainID >= noseHooverChains.size()) {
+        throw OpenMMException("Cannot set collision frequency for chainID " + std::to_string(chainID)
+                + ". Only " + std::to_string(noseHooverChains.size()) + " have been registered with this integrator."
+        );
+    }
+    noseHooverChains.at(chainID).setDefaultCollisionFrequency(frequency);
+}
+
+double NoseHooverIntegrator::getRelativeCollisionFrequency(int chainID) const {
+    if (chainID >= noseHooverChains.size()) {
+        throw OpenMMException("Cannot get relative collision frequency for chainID " + std::to_string(chainID)
+                + ". Only " + std::to_string(noseHooverChains.size()) + " have been registered with this integrator."
+        );
+    }
+    return noseHooverChains.at(chainID).getDefaultRelativeCollisionFrequency();
+}
+
+void NoseHooverIntegrator::setRelativeCollisionFrequency(double frequency, int chainID){
+    if (chainID >= noseHooverChains.size()) {
+        throw OpenMMException("Cannot set relative collision frequency for chainID " + std::to_string(chainID)
+                + ". Only " + std::to_string(noseHooverChains.size()) + " have been registered with this integrator."
+        );
+    }
+    noseHooverChains.at(chainID).setDefaultRelativeCollisionFrequency(frequency);
+}
+
+double NoseHooverIntegrator::computeKineticEnergy() {
+    double kE = 0.0;
+    if(noseHooverChains.size()) {
```

Comment:
Let's make this `if (noseHooverChains.size() > 0)`.  Maybe just a personal preference, but I find it clearer.